main.tf

```
@@ -0,0 +1,59 @@
+resource "google_pubsub_subscription_iam_binding" "binding" {
+  count = var.module_enabled && var.policy_bindings == null && var.authoritative ? 1 : 0
+
+  project = var.project
+
+  subscription = var.subscription
+
+  role    = var.role
+  members = var.members
+
+  depends_on = [var.module_depends_on]
+}
+
+resource "google_pubsub_subscription_iam_member" "member" {
+  for_each = var.module_enabled && var.policy_bindings == null && var.authoritative == false ? var.members : []
+
+  project = var.project
+
+  subscription = var.subscription
+
+  role   = var.role
+  member = each.value
+
+  depends_on = [var.module_depends_on]
+}
+
+resource "google_pubsub_subscription_iam_policy" "policy" {
+  count = var.module_enabled && var.policy_bindings != null ? 1 : 0
```

Policy bindings are authoritative, having then defined but setting authoritative to false should fail or not create the bindings.

```suggestion
  count = var.module_enabled && var.policy_bindings != null && var.authoritative ? 1 : 0
```

---

nope see other comment